carma_cooperative_perception/README.md

```diff
@@ -0,0 +1,32 @@
+# CARMA cooperative perception stack
+
+The `carma_cooperative_perception` package contains the cooperative perception stack used in CARMA Platform.
+It wraps the `cooperative_perception_core` library in a ROS 2 interface. The stack provides two main functionalities.
+The multiple object tracking functionality aggregates object data from incoming Sensor Data Sharing Messages (SDSMs),
+Basic Safety Messages (BSMs), and local sensors. The sensor data sharing functionality broadcasts SDSMs containing
+object data generated by the host’s local sensors. Parts of the stack can be disabled to support CARMA Platform
+vehicles with different sensor and computational capabilities. CARMA Platform deployments that include this package can
+collaborate with other agents that also share object tracking information to compose a cooperative perception network.
+
+The figures below show the general Node diagram and how the cooperative perception stack fits into the rest of the
+CARMA Platform architecture.
+
+> [!IMPORTANT]\
+> **Current limitations:** The package currently does not support BSMs.
+
+## CMake options
+
+The table below contains package-specific CMake options.
+
+| Option                                     | Default value      | Description                |
+| ------------------------------------------ | ------------------ | -------------------------- |
+| `carma_cooperative_perception_BUILD_TESTS` | `${BUILD_TESTING}` | Build the package's tests. |
```

Comment:
What is the purpose of having a package specific testing option/flag. Do other carma-platform packages have similar cmake options/flags? Do you see some benefit in having a specific flag for each package instead of just one flag that all packages share?

Reply:
This allows a user to specifically disable tests for this package if they want to. By default, `colcon` enables the `BUILD_TESTING` flag. It's a nice quality of life if you want to run tests for a specific package but don't care about any dependencies' tests. For example, they could do this

```shell
$ colcon build --packages-select my_package --cmake-args -DBUILD_TESTING=OFF -Dmy_package_BUILD_TESTS=ON
```

I recommend using this structure for all CARMA packages because it adds flexibility with minimal effort. It also follows the guidance from Craig Scott in his Professional CMake book.

Reply:
Sounds good to me. If you would like for all packages to have this feature you may have to in the future come up with some kind of how to guide for the rest of the team but that is out of scope for this issue.